From c0355890d52a845abe729f5db4d888467cdf1f2d Mon Sep 17 00:00:00 2001
From: Tom Rix <Tom.Rix@amd.com>
Date: Sun, 24 Nov 2024 06:27:07 -0800
Subject: [PATCH] Remove FromArrow functions

In the upstream arrow project, there is a api change
arrowArrow.Interface -> arrow.Array

Since ollama does not use these functions and they will problematic,
remove them.

Signed-off-by: Tom Rix <Tom.Rix@amd.com>
---
 dense_compat.go              | 179 -----------
 dense_compat_test.go         | 570 -----------------------------------
 example_dense_compat_test.go |  32 --
 go.mod                       |   1 -
 4 files changed, 782 deletions(-)
 delete mode 100644 example_dense_compat_test.go

diff --git a/dense_compat.go b/dense_compat.go
index dcbefa22bd75..cdf67d0a6013 100644
--- a/dense_compat.go
+++ b/dense_compat.go
@@ -8,10 +8,6 @@ import (
 	"math/cmplx"
 	"reflect"
 
-	arrow "github.com/apache/arrow/go/arrow"
-	arrowArray "github.com/apache/arrow/go/arrow/array"
-	"github.com/apache/arrow/go/arrow/bitutil"
-	arrowTensor "github.com/apache/arrow/go/arrow/tensor"
 	"github.com/chewxy/math32"
 	"github.com/pkg/errors"
 	"gonum.org/v1/gonum/mat"
@@ -438,178 +434,3 @@ func ToMat64(t *Dense, opts ...FuncOpt) (retVal *mat.Dense, err error) {
 	return
 }
 
-// FromArrowArray converts an "arrow/array".Interface into a Tensor of matching DataType.
-func FromArrowArray(a arrowArray.Interface) *Dense {
-	a.Retain()
-	defer a.Release()
-
-	r := a.Len()
-
-	// TODO(poopoothegorilla): instead of creating bool ValidMask maybe
-	// bitmapBytes can be used from arrow API
-	mask := make([]bool, r)
-	for i := 0; i < r; i++ {
-		mask[i] = a.IsNull(i)
-	}
-
-	switch a.DataType() {
-	case arrow.BinaryTypes.String:
-		backing := make([]string, r)
-		for i := 0; i < r; i++ {
-			backing[i] = a.(*arrowArray.String).Value(i)
-		}
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.FixedWidthTypes.Boolean:
-		backing := make([]bool, r)
-		for i := 0; i < r; i++ {
-			backing[i] = a.(*arrowArray.Boolean).Value(i)
-		}
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Int8:
-		backing := a.(*arrowArray.Int8).Int8Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Int16:
-		backing := a.(*arrowArray.Int16).Int16Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Int32:
-		backing := a.(*arrowArray.Int32).Int32Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Int64:
-		backing := a.(*arrowArray.Int64).Int64Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Uint8:
-		backing := a.(*arrowArray.Uint8).Uint8Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Uint16:
-		backing := a.(*arrowArray.Uint16).Uint16Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Uint32:
-		backing := a.(*arrowArray.Uint32).Uint32Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Uint64:
-		backing := a.(*arrowArray.Uint64).Uint64Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Float32:
-		backing := a.(*arrowArray.Float32).Float32Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	case arrow.PrimitiveTypes.Float64:
-		backing := a.(*arrowArray.Float64).Float64Values()
-		retVal := New(WithBacking(backing, mask), WithShape(r, 1))
-		return retVal
-	default:
-		panic(fmt.Sprintf("Unsupported Arrow DataType - %v", a.DataType()))
-	}
-
-	panic("Unreachable")
-}
-
-// FromArrowTensor converts an "arrow/tensor".Interface into a Tensor of matching DataType.
-func FromArrowTensor(a arrowTensor.Interface) *Dense {
-	a.Retain()
-	defer a.Release()
-
-	if !a.IsContiguous() {
-		panic("Non-contiguous data is Unsupported")
-	}
-
-	var shape []int
-	for _, val := range a.Shape() {
-		shape = append(shape, int(val))
-	}
-
-	l := a.Len()
-	validMask := a.Data().Buffers()[0].Bytes()
-	dataOffset := a.Data().Offset()
-	mask := make([]bool, l)
-	for i := 0; i < l; i++ {
-		mask[i] = len(validMask) != 0 && bitutil.BitIsNotSet(validMask, dataOffset+i)
-	}
-
-	switch a.DataType() {
-	case arrow.PrimitiveTypes.Int8:
-		backing := a.(*arrowTensor.Int8).Int8Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Int16:
-		backing := a.(*arrowTensor.Int16).Int16Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Int32:
-		backing := a.(*arrowTensor.Int32).Int32Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Int64:
-		backing := a.(*arrowTensor.Int64).Int64Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Uint8:
-		backing := a.(*arrowTensor.Uint8).Uint8Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Uint16:
-		backing := a.(*arrowTensor.Uint16).Uint16Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Uint32:
-		backing := a.(*arrowTensor.Uint32).Uint32Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Uint64:
-		backing := a.(*arrowTensor.Uint64).Uint64Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Float32:
-		backing := a.(*arrowTensor.Float32).Float32Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	case arrow.PrimitiveTypes.Float64:
-		backing := a.(*arrowTensor.Float64).Float64Values()
-		if a.IsColMajor() {
-			return New(WithShape(shape...), AsFortran(backing, mask))
-		}
-
-		return New(WithShape(shape...), WithBacking(backing, mask))
-	default:
-		panic(fmt.Sprintf("Unsupported Arrow DataType - %v", a.DataType()))
-	}
-
-	panic("Unreachable")
-}
diff --git a/dense_compat_test.go b/dense_compat_test.go
index c641203489c7..1f7feaebeaff 100644
--- a/dense_compat_test.go
+++ b/dense_compat_test.go
@@ -5,10 +5,6 @@ package tensor
 import (
 	"testing"
 
-	arrow "github.com/apache/arrow/go/arrow"
-	arrowArray "github.com/apache/arrow/go/arrow/array"
-	"github.com/apache/arrow/go/arrow/memory"
-	arrowTensor "github.com/apache/arrow/go/arrow/tensor"
 	"github.com/stretchr/testify/assert"
 	"gonum.org/v1/gonum/mat"
 )
@@ -110,569 +106,3 @@ func TestFromMat64(t *testing.T) {
 	}
 }
 
-var toArrowArrayTests = []struct {
-	data  interface{}
-	valid []bool
-	dt    arrow.DataType
-	shape Shape
-}{
-	{
-		data:  Range(Int8, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Int8,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Int16, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Int16,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Int32, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Int32,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Int64, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Int64,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Uint8, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Uint8,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Uint16, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Uint16,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Uint32, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Uint32,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Uint64, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Uint64,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Float32, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Float32,
-		shape: Shape{6, 1},
-	},
-	{
-		data:  Range(Float64, 0, 6),
-		valid: []bool{true, true, true, false, true, true},
-		dt:    arrow.PrimitiveTypes.Float64,
-		shape: Shape{6, 1},
-	},
-}
-
-func TestFromArrowArray(t *testing.T) {
-	assert := assert.New(t)
-	var T *Dense
-	pool := memory.NewGoAllocator()
-
-	for i, taat := range toArrowArrayTests {
-		var m arrowArray.Interface
-
-		switch taat.dt {
-		case arrow.BinaryTypes.String:
-			b := arrowArray.NewStringBuilder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]string{"0", "1", "2", "3", "4", "5"},
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.FixedWidthTypes.Boolean:
-			b := arrowArray.NewBooleanBuilder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]bool{true, false, true, false, true, false},
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Int8:
-			b := arrowArray.NewInt8Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Int8, 0, 6).([]int8),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Int16:
-			b := arrowArray.NewInt16Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Int16, 0, 6).([]int16),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Int32:
-			b := arrowArray.NewInt32Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Int32, 0, 6).([]int32),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Int64:
-			b := arrowArray.NewInt64Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Int64, 0, 6).([]int64),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Uint8:
-			b := arrowArray.NewUint8Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Uint8, 0, 6).([]uint8),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Uint16:
-			b := arrowArray.NewUint16Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Uint16, 0, 6).([]uint16),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Uint32:
-			b := arrowArray.NewUint32Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Uint32, 0, 6).([]uint32),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Uint64:
-			b := arrowArray.NewUint64Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Uint64, 0, 6).([]uint64),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Float32:
-			b := arrowArray.NewFloat32Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Float32, 0, 6).([]float32),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		case arrow.PrimitiveTypes.Float64:
-			b := arrowArray.NewFloat64Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				Range(Float64, 0, 6).([]float64),
-				taat.valid,
-			)
-			m = b.NewArray()
-			defer m.Release()
-		default:
-			t.Errorf("DataType not supported in tests: %v", taat.dt)
-		}
-
-		T = FromArrowArray(m)
-		switch taat.dt {
-		case arrow.PrimitiveTypes.Int8:
-			conv := taat.data.([]int8)
-			assert.Equal(conv, T.Int8s(), "test %d: []int8 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Int16:
-			conv := taat.data.([]int16)
-			assert.Equal(conv, T.Int16s(), "test %d: []int16 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Int32:
-			conv := taat.data.([]int32)
-			assert.Equal(conv, T.Int32s(), "test %d: []int32 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Int64:
-			conv := taat.data.([]int64)
-			assert.Equal(conv, T.Int64s(), "test %d: []int64 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Uint8:
-			conv := taat.data.([]uint8)
-			assert.Equal(conv, T.Uint8s(), "test %d: []uint8 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Uint16:
-			conv := taat.data.([]uint16)
-			assert.Equal(conv, T.Uint16s(), "test %d: []uint16 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Uint32:
-			conv := taat.data.([]uint32)
-			assert.Equal(conv, T.Uint32s(), "test %d: []uint32 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Uint64:
-			conv := taat.data.([]uint64)
-			assert.Equal(conv, T.Uint64s(), "test %d: []uint64 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Float32:
-			conv := taat.data.([]float32)
-			assert.Equal(conv, T.Float32s(), "test %d: []float32 from %v", i, taat.dt)
-		case arrow.PrimitiveTypes.Float64:
-			conv := taat.data.([]float64)
-			assert.Equal(conv, T.Float64s(), "test %d: []float64 from %v", i, taat.dt)
-		default:
-			t.Errorf("DataType not supported in tests: %v", taat.dt)
-		}
-		for i, invalid := range T.Mask() {
-			assert.Equal(taat.valid[i], !invalid)
-		}
-		assert.True(T.Shape().Eq(taat.shape))
-	}
-}
-
-var toArrowTensorTests = []struct {
-	rowMajorData  interface{}
-	colMajorData  interface{}
-	rowMajorValid []bool
-	colMajorValid []bool
-	dt            arrow.DataType
-	shape         Shape
-}{
-	{
-		rowMajorData:  []int8{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []int8{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Int8,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []int16{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []int16{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Int16,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []int32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []int32{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Int32,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []int64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []int64{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Int64,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []uint8{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []uint8{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Uint8,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []uint16{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []uint16{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Uint16,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []uint32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []uint32{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Uint32,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []uint64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []uint64{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Uint64,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []float32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []float32{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Float32,
-		shape:         Shape{2, 5},
-	},
-	{
-		rowMajorData:  []float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		colMajorData:  []float64{1, 6, 2, 7, 3, 8, 4, 9, 5, 10},
-		rowMajorValid: []bool{true, false, true, false, true, false, true, false, true, false},
-		colMajorValid: []bool{true, false, false, true, true, false, false, true, true, false},
-		dt:            arrow.PrimitiveTypes.Float64,
-		shape:         Shape{2, 5},
-	},
-}
-
-func TestFromArrowTensor(t *testing.T) {
-	assert := assert.New(t)
-	var rowMajorT *Dense
-	var colMajorT *Dense
-	pool := memory.NewGoAllocator()
-
-	for i, taat := range toArrowTensorTests {
-		var rowMajorArr arrowArray.Interface
-		var colMajorArr arrowArray.Interface
-		var rowMajor arrowTensor.Interface
-		var colMajor arrowTensor.Interface
-
-		switch taat.dt {
-		case arrow.PrimitiveTypes.Int8:
-			b := arrowArray.NewInt8Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]int8{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]int8{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewInt8(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewInt8(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Int8SizeBytes), int64(arrow.Int8SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Int16:
-			b := arrowArray.NewInt16Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]int16{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]int16{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewInt16(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewInt16(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Int16SizeBytes), int64(arrow.Int16SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Int32:
-			b := arrowArray.NewInt32Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]int32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]int32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewInt32(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewInt32(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Int32SizeBytes), int64(arrow.Int32SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Int64:
-			b := arrowArray.NewInt64Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]int64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]int64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewInt64(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewInt64(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Int64SizeBytes), int64(arrow.Int64SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Uint8:
-			b := arrowArray.NewUint8Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]uint8{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]uint8{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewUint8(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewUint8(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Uint8SizeBytes), int64(arrow.Uint8SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Uint16:
-			b := arrowArray.NewUint16Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]uint16{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]uint16{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewUint16(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewUint16(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Uint16SizeBytes), int64(arrow.Uint16SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Uint32:
-			b := arrowArray.NewUint32Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]uint32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]uint32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewUint32(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewUint32(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Uint32SizeBytes), int64(arrow.Uint32SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Uint64:
-			b := arrowArray.NewUint64Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]uint64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]uint64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewUint64(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewUint64(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Uint64SizeBytes), int64(arrow.Uint64SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Float32:
-			b := arrowArray.NewFloat32Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]float32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]float32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewFloat32(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewFloat32(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Float32SizeBytes), int64(arrow.Float32SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		case arrow.PrimitiveTypes.Float64:
-			b := arrowArray.NewFloat64Builder(pool)
-			defer b.Release()
-			b.AppendValues(
-				[]float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			rowMajorArr = b.NewArray()
-			defer rowMajorArr.Release()
-
-			b.AppendValues(
-				[]float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-				taat.rowMajorValid,
-			)
-			colMajorArr = b.NewArray()
-			defer colMajorArr.Release()
-
-			rowMajor = arrowTensor.NewFloat64(rowMajorArr.Data(), []int64{2, 5}, nil, []string{"x", "y"})
-			defer rowMajor.Release()
-			colMajor = arrowTensor.NewFloat64(colMajorArr.Data(), []int64{2, 5}, []int64{int64(arrow.Float64SizeBytes), int64(arrow.Float64SizeBytes * 2)}, []string{"x", "y"})
-			defer colMajor.Release()
-		default:
-			t.Errorf("DataType not supported in tests: %v", taat.dt)
-		}
-
-		rowMajorT = FromArrowTensor(rowMajor)
-		colMajorT = FromArrowTensor(colMajor)
-
-		assert.Equal(taat.rowMajorData, rowMajorT.Data(), "test %d: row major %v", i, taat.dt)
-		assert.Equal(len(taat.rowMajorValid), len(rowMajorT.Mask()), "test %d: row major %v mask length incorrect", i, taat.dt)
-		for i, invalid := range rowMajorT.Mask() {
-			assert.Equal(taat.rowMajorValid[i], !invalid, "test %d: row major %v mask value incorrect", i, taat.dt)
-		}
-		assert.True(colMajorT.Shape().Eq(taat.shape))
-
-		assert.Equal(taat.colMajorData, colMajorT.Data(), "test %d: column major %v", i, taat.dt)
-		assert.Equal(len(taat.colMajorValid), len(colMajorT.Mask()), "test %d: column major %v mask length incorrect", i, taat.dt)
-		for i, invalid := range colMajorT.Mask() {
-			assert.Equal(taat.colMajorValid[i], !invalid, "test %d: column major %v mask value incorrect", i, taat.dt)
-		}
-		assert.True(rowMajorT.Shape().Eq(taat.shape))
-	}
-}
diff --git a/example_dense_compat_test.go b/example_dense_compat_test.go
deleted file mode 100644
index 22d7c3a85041..000000000000
--- a/example_dense_compat_test.go
+++ /dev/null
@@ -1,32 +0,0 @@
-package tensor_test
-
-import (
-	"fmt"
-
-	"github.com/apache/arrow/go/arrow/array"
-	"github.com/apache/arrow/go/arrow/memory"
-	"github.com/pdevine/tensor"
-)
-
-func ExampleDense_Arrow() {
-	pool := memory.NewGoAllocator()
-
-	b := array.NewFloat64Builder(pool)
-	defer b.Release()
-
-	b.AppendValues(
-		[]float64{1, 2, 3, -1, 4, 5},
-		[]bool{true, true, true, false, true, true},
-	)
-
-	arr := b.NewFloat64Array()
-	defer arr.Release()
-	fmt.Printf("arrow array = %v\n", arr)
-
-	a := tensor.FromArrowArray(arr)
-	fmt.Printf("tensor = %v\n", a)
-
-	// Output:
-	// arrow array = [1 2 3 (null) 4 5]
-	// tensor = C[ 1   2   3  --   4   5]
-}
diff --git a/go.mod b/go.mod
index 410feab9fc89..63876cfe58c2 100644
--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,6 @@ module github.com/pdevine/tensor
 go 1.22.0
 
 require (
-	github.com/apache/arrow/go/arrow v0.0.0-20211112161151-bc219186db40
 	github.com/chewxy/hm v1.0.0
 	github.com/chewxy/math32 v1.10.1
 	github.com/gogo/protobuf v1.3.2
-- 
2.47.0

